monitoring/uss_qualifier/suites/faa/uft/design/README.md

```diff
@@ -0,0 +1,87 @@
+
+1. Missing message signing headers - The first step of message signing analysis is to check for the existence of all of the required message signing headers. These headers are `content-digest`, `utm-message-signature`, `utm-message-signature-input`, and `x-utm-jws-header`. If any of these headers are missing in incoming requests to the mock_uss, or are missing in received responses from the USS under test, then that will generate an issue.
+2. Invalid content digests - Once all message signing headers are accounted for, the sha-512 hash on either the request/response body (depending on what is being analyzed), is taken, and this value is compared to the one received in the `content-digest` header. If these values differ, this will generate an issue showing the difference in the values. 
+3. Invalid signature - After the content-digest check is performed, the `utm-message-signature` is analyzed to ensure that it is a valid signature. To do this check, the signature base is created, and the public key served in the `x5u` field within the `x-utm-jws-header` is retreived. Using the value in the `utm-message-signature` header field, the public key, and the recreated signature base, it is determined whether or not the signature is valid.
+
+
+### Positive tests -
+A valid set of private/public keys are provided for message signing analysis under the [monitioring/monitorlib/messagesigning/keys] folder. The valid key pair, `mock_faa_priv.pem`/`mock_faa_pub.der`, is used by default. The public key is served under the mock_uss endpoint `/mock/scd/.well-known/uas-traffic-management/mock_pub.der`, and can be retreived by the USS under test in order for it to validate the mock_uss responses. This public key was provided by the FAA and will pass SCVP validation. To ensure that this valid keypair is used, make sure that the `USE_VALID_KEY_PAIR` in `run_locally_msgsigning.sh` is set to "true". 
+
+
+A USS should pass all the uss_qualifier tests in this suite.
+As well as the message_signing report should have no issues reported. A USS will provide both of these reports.
+The message_signing report includes interactions and issues between the mock_uss and the USS-under-test.
+No issues indicate the USS-under-test message-signed all its requests and responses.
+
+
+### Negative tests -
+Replace the private/public keys with invalid key pair by setting the `USE_VALID_KEY_PAIR` in `run_locally_msgsigning.sh` to "false". This will set the keypair to be `mock_priv.pem`/`mock_pub.der`. Using this keypair will lead to invalid signatures, and the `mock_pub.der` will not pass SCVP validation.
+The USS-under-test will respond with 403 to all requests from mock_uss. The Uss Qualifier tests will not pass.
+The message signing section of the report would show `403` in interactions with mock_uss.
```

Comment:
```suggestion
### Negative tests
In the future, this test suite will be expanded to instruct mock_uss to replace the private/public keys with an invalid key pair by calling the appropriate endpoint. This will set the keypair to be `mock_priv.pem`/`mock_pub.der`. Using this keypair will lead to invalid signatures on mock_uss's interactions, and the `mock_pub.der` will not pass SCVP validation.
The USS-under-test should respond with 403 to all requests from mock_uss. The Uss Qualifier tests will require this to happen and only pass if the correct behavior is demonstrated.
The message signing section of the report would show `403` in interactions with mock_uss.
```